src/app/Core/ExportServices/generateDataString.service.ts

```diff
@@ -7,13 +46,18 @@ import { generateCSVString } from "./generateCSVString.service";
  * @returns {string} - dataset as string
  */
 export function generateDataString(
-    keys: string[],
-    data: Record<string, string>[]
+    data: DbCollection,
+    keys: DbKey[], 
+    mainFilterLabels: MainFilterLabelInterface,
+    mainFilterAttributes: string[],
 ): string {
     const FilteredDataString: string[] = [];
-    const headers: string[] = keys;
-    FilteredDataString.push(headers.join(","));
-    FilteredDataString.push(generateCSVString(data, headers));
+    const headerArray: string[] = []
+    mainFilterAttributes.forEach(element => {
+        headerArray.push(mainFilterLabels[element])
+    });

```

Comment:
could be simplified: headerArray = mainFilterAttributes.map(element =>mainFiltrrLabels[element])

Reply:
Done. Commit "Simplify code"